src/main/resources/migration/changelog-v5.6.0.xml

```diff
@@ -107,4 +107,202 @@
+            </column>
+            <column name="ALGORITHM_PROPERTY_ID" type="BIGINT"/>
+            <column name="CRYPTO_FUNCTION" type="VARCHAR(32)"/>
+        </createTable>
+
+        <createTable tableName="PROTOCOL_CIPHER_SUITES">
+            <column name="PROTOCOL_PROPERTY_ID" type="BIGINT"/>
+            <column name="CIPHER_SUITE_ID" type="BIGINT"/>
+        </createTable>
+
+        <createTable tableName="CIPHER_SUITE">
+            <column autoIncrement="true" name="ID" type="BIGINT">
+                <constraints nullable="false" primaryKey="true" primaryKeyName="CIPHER_SUITE_PK"/>
+            </column>
+            <column name="NAME" type="VARCHAR(64)"/>
+        </createTable>
+
+        <createTable tableName="CIPHER_SUITE_ALGORITHM">
+            <column name="CIPHER_SUITE_ID" type="BIGINT"/>
+            <column name="ALGORITHM" type="VARCHAR(64)"/>
+        </createTable>
+
+        <createTable tableName="CIPHER_SUITE_IDENTIFIER">
+            <column name="CIPHER_SUITE_ID" type="BIGINT"/>
+            <column name="IDENTIFIER" type="VARCHAR(8)"/>
+        </createTable>
+
+        <createTable tableName="PROTOCOL_IKEV2_TYPES">
+            <column name="PROTOCOL_PROPERTY_ID" type="BIGINT"/>
+            <column name="IKEV2_TYPE_ID" type="BIGINT"/>
+        </createTable>
+
+        <createTable tableName="IKEV2_TYPE">
+            <column autoIncrement="true" name="ID" type="BIGINT">
+                <constraints nullable="false" primaryKey="true" primaryKeyName="IKEV2_TYPE_PK"/>
+            </column>
+            <column name="TYPE" type="VARCHAR(16)"/>
+        </createTable>
+
+        <createTable tableName="IKEV2_REF">
+            <column name="IKEV2_TYPE_ID" type="BIGINT"/>
+            <column name="REF" type="VARCHAR(64)"/>
+        </createTable>
+
+        <createTable tableName="CRYPTO_REFS">
+            <column autoIncrement="true" name="ID" type="BIGINT">
+                <constraints nullable="false" primaryKey="true" primaryKeyName="CRYPTO_REF_PK"/>
+            </column>
+            <column name="PROTOCOL_PROPERTY_ID" type="BIGINT"/>
+            <column name="CRYPTO_REF" type="VARCHAR(64)"/>
+        </createTable>
+
+        <createTable tableName="COMPONENT_OCCURRENCES">
+            <column name="COMPONENT_ID" type="BIGINT"/>
+            <column name="OCCURRENCE_ID" type="BIGINT"/>
+        </createTable>
+
+        <createTable tableName="OCCURRENCE">
+            <column autoIncrement="true" name="ID" type="BIGINT">
+                <constraints nullable="false" primaryKey="true" primaryKeyName="OCCURRENCES_PK"/>
+            </column>
+            <column name="BOM_REF" type="VARCHAR(64)"/>
+            <column name="LOCATION" type="VARCHAR(255)"/>
+            <column name="LINE" type="INT"/>
+            <column name="OFFSET" type="INT"/>
+            <column name="SYMBOL" type="INT"/>
+            <column name="ADDITIONAL_CONTEXT" type="VARCHAR(255)"/>
```

Comment:
Hi @nscuro, thanks a lot for your review.

The code above comes from the definition of the OCCURRENCES table.  The PR currently adds all crypto-related data fields that our scanner (https://github.com/IBM/cbomkit) generates. We can drop OCCURRENCES because the location of crypto assets in source is probably not important in DT.

Nevertheless, the your point re. the size of the varchar columns is still valid. There is no proper reasoning behind the current numbers. We started with some guesstimates based on standard (https://cyclonedx.org/docs/1.6/json/) and examples (https://github.com/CycloneDX/bom-examples/tree/master/CBOM) and simply doubled if the initial numbers weren't sufficient. I'd gladly use other numbers with better foundation. If data size is an issue we should go through all varchars one-by-one and decide (a) does it have to be persisted and (b) if so, what size is needed.

Reply:
> We can drop OCCURRENCES because the location of crypto assets in source is probably not important in DT.

Occurrence will likely come to DT in the future to cater to other use cases, but if it's not crucial for the crypto functionality it's good to drop here IMO.

> I'd gladly use other numbers with better foundation. If data size is an issue we should go through all varchars one-by-one and decide (a) does it have to be persisted and (b) if so, what size is needed.

In PostgreSQL it's all `TEXT` (unlimited length) behind the scenes, with an implicit `CHECK` constraint for the length. So other than preventing users from storing GBs of useless data, size constraints don't matter. With this in mind, I think choosing an initial length of `1024` should be fine.

Reply:
> Occurrence will likely come to DT in the future to cater to other use cases, but if it's not crucial for the crypto functionality it's good to drop here IMO
>....
> With this in mind, I think choosing an initial length of 1024 should be fine.



Okay, thanks.
1. I will remove Occurrence with the next commit.
2. VARCHAR length will default to 1024 unless we know better.